### ISSUE TITLE
How is emitting \r to stdout a good plan?

### ISSUE THREAD

**Issue by jernst** [aws/aws-cli#3366]
Here is my transcript:
```
% echo hello > hello
% aws s3 cp hello s3://indiecomp-test/hello > out
% cat out 
upload: ./hello to s3://indiecomp-test/hello     
% # So far, so good, but now:             
% od -c out
0000000   C   o   m   p   l   e   t   e   d       6       B   y   t   e
0000020   s   /   6       B   y   t   e   s       (   5       B   y   t
0000040   e   s   /   s   )       w   i   t   h       1       f   i   l
0000060   e   (   s   )       r   e   m   a   i   n   i   n   g  \r   u
0000100   p   l   o   a   d   :       .   /   h   e   l   l   o       t
0000120   o       s   3   :   /   /   i   n   d   i   e   c   o   m   p
0000140   -   t   e   s   t   /   h   e   l   l   o                    
0000160                                                      \n
0000176
```
There is an entire message being printed, which is overwritten within the next millisecond or so, and thus is never seen by the user. Which makes for very frustrating script programming!

But even once one has discovered what is going on here, it makes it really hard to parse the output of this command, and from what I can tell of the options, this cannot be turned off. (`--out json` does not seem to do anything).

At the very minimum, provide an option to turn off those messages like `curl --silent`, but much better would be to write progress messages to stderr, not stdout, also like curl.

**Comment by kyleknap** [comment]
There are a few options to control the output as shown in the [docs](https://docs.aws.amazon.com/cli/latest/reference/s3/cp.html):
* `--quiet`: It will remove all output (i.e. progress, results, errors)
* `--no-progress`: It will remove the progress entire and only show results and errors. It sounds like that this will probably best fit the bill here because it will not use return carriages.
* `--only-show-errors`: This will only display errors

Let us know if one of these options work for you.

**Comment by jernst** [comment]
Oh, sure, as does my modified regex once `od` told me what was going on. The problem is that your default isn't reasonable from the perspective of how a command-line program is supposed to behave, costing developers a bunch of time and aggrevation.

**Comment by JordonPhillips** [comment]
The intent is to enhance interactive use by showing progress in the terminal. Using `\r` is the standard cross-platform way to wipe a line, allowing us to update percentages and transfer speeds without filling up the terminal. This can get annoying when redirecting output though, which is why we also provide controls to eliminate that behavior.

**Comment by jernst** [comment]
Seems to me there are much better ways to do that, e.g. http://stackoverflow.com/a/911213/1219634. No flags required.

**Comment by justnance** [comment]
@jernst - Thanks for your feedback.  At this time we are not looking to change the behavior as we need to preserve backwards compatibility.

**Comment by jernst** [comment]
Do you have a specific use case for this particular type of backwards compatibility requirement? That would be unusual, if you implement what's documented on that Stackoverflow link.
